src/lib/core/CHIPConfig.h

```diff
@@ -991,7 +1002,11 @@ extern const char CHIP_NON_PRODUCTION_MARKER[];
  * Binds to number of GroupState entries to support per fabric
  */
 #ifndef CHIP_CONFIG_MAX_GROUPS_PER_FABRIC
-#define CHIP_CONFIG_MAX_GROUPS_PER_FABRIC 3
+#define CHIP_CONFIG_MAX_GROUPS_PER_FABRIC (4 * CHIP_CONFIG_MAX_GROUP_ENDPOINTS_PER_FABRIC)
+#endif
+
+#if CHIP_CONFIG_MAX_GROUPS_PER_FABRIC < (4 * CHIP_CONFIG_MAX_GROUP_ENDPOINTS_PER_FABRIC)
```

Comment:
Need to account for when groups are not enabled at all. The value 0 should be OK

Reply:
If CHIP_CONFIG_MAX_GROUPS_PER_FABRIC is 0, then CHIP_CONFIG_MAX_GROUP_ENDPOINTS_PER_FABRIC must also be 0, so this will not produce an error.